pkg/rngd/cmd/rngd/main.go

```diff
@@ -81,16 +155,29 @@ type randInfo struct {
 }
 
 func writeEntropy(random *os.File) (int, error) {
-	r, err := rand()
-	if err != nil {
-		// assume can fail occasionally
-		return 0, nil
-	}
-	const entropy = 64 // they are good random numbers, Brent
-	info := randInfo{entropy, 8, r}
-	ret, _, err := unix.Syscall(unix.SYS_IOCTL, uintptr(random.Fd()), uintptr(C.rndaddentropy), uintptr(unsafe.Pointer(&info)))
-	if ret == 0 {
-		return 8, nil
+	var i int
+	var r uint64
+	var err error
+	var ret uintptr
+	for _, e := range entropySources {
```

Comment:
So, it takes from all enabled sources?

Reply:
Yes, The more entropy sources available, the more better quality of the entropies in the pool, the more safer to use it.